technic/machines/switching_station.lua

```diff
@@ -202,14 +209,48 @@ local traverse_network = function(PR_nodes, RE_nodes, BA_nodes, SP_nodes, all_no
 	end
 end
 
-local touch_nodes = function(list, tier)
+function technic.pos2network(pos)
+	return pos and technic.cables[minetest.hash_node_position(pos)]
+end
+
+function technic.network2pos(network_id)
+	return network_id and minetest.get_position_from_hash(network_id)
+end
+
+function technic.network2sw_pos(network_id)
+	-- Return switching station position for network.
+	-- It is not guaranteed that position actually contains switching station.
+	local sw_pos = minetest.get_position_from_hash(network_id)
+	sw_pos.y = sw_pos.y + 1
+	return sw_pos
+end
```

Comment:
These could be better defined in some common helper/utility file instead or even bettersome network.lua but there's no such thing, at least not yet.